FordLabs/retroquest
# Fix sorting on empty column bug, Finish refactoring to new column header

component

[#44, #45]

## Overview
This PR fixes a sorting bug that was caused by sorting a column when no thoughts were added. This caused a splicing error on the internal array.

### Demo
The code looks pretty much the same visually as it did before, only some minor tweaks were added.

## Testing Instructions
Open up the developer console when you have the application running and sort on a column that has no thoughts added. You should see no error logs.
